app/uk/gov/hmrc/gform/gform/handlers/FormControllerRequestHandler.scala

```diff
@@ -54,17 +61,52 @@ class FormControllerRequestHandler(formValidator: FormValidator)(implicit ec: Ex
     cache: CacheData,
     envelope: EnvelopeWithMapping,
     validatePageModel: ValidatePageModel[Future, DataOrigin.Browser],
-    enteredVariadicFormData: EnteredVariadicFormData
-  ): Future[FormValidationOutcome] =
-    formValidator
-      .validatePageModelBySectionNumber(
-        formModelOptics,
-        sectionNumber,
-        cache,
-        envelope,
-        validatePageModel
+    enteredVariadicFormData: EnteredVariadicFormData,
+    form: Form,
+    retrievals: MaterialisedRetrievals
+  )(implicit hc: HeaderCarrier, lang: LangADT): Future[FormValidationOutcome] =
+    for {
+      formHandlerResult <- formValidator.validatePageModelBySectionNumber(
+                             formModelOptics,
+                             sectionNumber,
+                             cache,
+                             envelope,
+                             validatePageModel
+                           )
+
+      outcome = formValidator.toFormValidationOutcome(formHandlerResult, enteredVariadicFormData)
+
+      _ <- if (!outcome.isValid) {
+             auditFormValidationErrors(formHandlerResult, form, retrievals)
+           } else {
+             Future.successful(())
+           }
+
+    } yield outcome
+
+  private def auditFormValidationErrors(
+    formHandlerResult: FormHandlerResult,
+    form: Form,
+    retrievals: MaterialisedRetrievals
+  )(implicit hc: HeaderCarrier, lang: LangADT): Future[Unit] = {
+    val validationErrors: Map[FormComponentId, List[String]] =
+      formHandlerResult.validationResult.lookup.collect {
+        case (fcId, ffvr) if ffvr.fieldErrors.nonEmpty =>
+          fcId -> ffvr.fieldErrors.toList
+      }
+
+    if (validationErrors.nonEmpty) {
+      Future.successful(
```

Comment:
I think this here to run `auditService.sendFormValidationErrorEvent` in non blocking fashion, right?

If so I think it should `Future` should be used instead of `Future.successful`, wdyt?